### PR TITLE
In get_params_from make mosaic_dir mandatory and add the exclude parameter

### PR DESCRIPTION
**Pull request by ptormene**
Reverting https://github.com/gem/oq-engine/pull/9328 and implementing the same functionality in an alternative way.

### CODE REVIEW

File: openquake/commands/mosaic.py [hunk 1]
```diff
@@ -105,7 +107,9 @@ def from_file(fname, concurrent_jobs=8):
                 continue
             dic = dict(siteid=siteid, lon=float(lon), lat=float(lat))
             tags.append(siteid)
-            allparams.append(get_params_from(dic))
+            # FIXME: should we exclude USA here or not?
```
Comment (micheles) [inline openquake/commands/mosaic.py:110]: The CSV file does not contain sites in USA (was built so) so there is not need to exclude the USA.

File: openquake/commands/mosaic.py [hunk 2]
```diff
@@ -143,7 +147,9 @@ def run_site(lonlat_or_fname, *, hc: int = None, slowest: int = None,
         from_file(lonlat_or_fname, concurrent_jobs)
         return
     lon, lat = lonlat_or_fname.split(',')
-    params = get_params_from(dict(lon=lon, lat=lat, vs30=vs30))
+    params = get_params_from(
+        dict(lon=lon, lat=lat, vs30=vs30), config.directory.mosaic_dir,
```
Comment (micheles) [inline openquake/commands/mosaic.py:151]: Not here, I want to be able to use `oq mosaic run_site` also in the USA for non-AELO calculations

File: openquake/engine/aelo.py [hunk 3]
```diff
@@ -112,39 +111,40 @@ def main(lon: valid.longitude,
...
-            # in  testing mode create a new job context
-            config.directory.mosaic_dir = os.path.join(
-                os.path.dirname(CDIR), 'qa_tests_data/mosaic')
-            dic = dict(calculation_mode='custom', description='AELO')
-            [jobctx] = engine.create_jobs([dic], config.distribution.log_level,
-                                          None, getpass.getuser(), None)
+    inputs = dict(lon=lon, lat=lat, vs30=vs30, siteid=siteid)
+    if jobctx is None:
+        # in  testing mode create a new job context
+        config.directory.mosaic_dir = os.path.join(
```
Comment (micheles) [inline openquake/engine/aelo.py:117]: No need to override the configuration object now, just pass the `mosaic_dir` downstream

File: openquake/server/views.py [hunk 4]
```diff
@@ -603,6 +603,9 @@ def aelo_run(request):
     :param request:
         a `django.http.HttpRequest` object containing lon, lat, vs30, siteid
     """
+    if not config.directory.mosaic_dir:
+        return HttpResponseServerError(
+            'mosaic_dir is not specified in openquake.cfg')
```
Comment (micheles) [inline openquake/server/views.py:608]: No, this is not the right place. If `mosaic_dir` is not set the WebServer must not start at all.

**Comment by micheles** [comment]
I wonder about the lines
```
PRELIMINARY_MODELS = ['CEA', 'CHN', 'NEA']
PRELIMINARY_MODEL_WARNING = (
    'Results are preliminary. The seismic hazard model used for the site'
    ' is under review and will be updated' ' during Year 3.')
```
How the warnings appear to the user? Are they really important? If not, we could just remove the call to `geo.utils.geolocate` in `get_params_from` with a good simplification. In any case, having `geo.utils.geolocate` inside `get_params_from`  looks like the wrong place.

**Comment by ptormene** [comment]
> I wonder about the lines
> 
> ```
> PRELIMINARY_MODELS = ['CEA', 'CHN', 'NEA']
> PRELIMINARY_MODEL_WARNING = (
>     'Results are preliminary. The seismic hazard model used for the site'
>     ' is under review and will be updated' ' during Year 3.')
> ```
> 
> How the warnings appear to the user? Are they really important?

This warning is sent via email and it was recently requested.

**Pull request merged**